### DISCUSSION TITLE
What does the "System" role do?

### DISCUSSION THREAD

**Discussion by TarekAS** [zed-industries/zed#16052] [Feature Requests]
When using the Assistant panel, there are 3 roles: "You", "Assistant" and "System". I understand the first 2 as they are properly documented, but what does "System" do?

**Comment by bajrangCoder** [comment]
The **system** role is used to set the context, rules, or instructions that guide how the AI (acting as the assistant) should behave during the interaction.

**Reply by horvathsanyi** [reply]
And what is the purpose of the assistant role? When do i want to use it? Thanks!
